### ISSUE TITLE
Customizing @material-ui TextWidget requires me to customize ALL of the TextWidget wrapped widgets

### ISSUE THREAD

**Issue by heath-freenome** [rjsf-team/react-jsonschema-form#2512]
### Prerequisites

- [ x] I have read the [documentation](https://react-jsonschema-form.readthedocs.io/);
- [ x] In the case of a bug report, I understand that providing a [SSCCE](http://sscce.org/) example is tremendously useful to the maintainers.
- [ x] Ideally, I'm providing a [sample JSFiddle](https://jsfiddle.net/n1k0/f2y3fq7L/6/) or a [shared playground link](https://rjsf-team.github.io/react-jsonschema-form/) demonstrating the issue.

### Description

Unlike the `core` extensions of the `BaseInput` which pull the `BaseInput` component out of the `registry.widgets` object, the ones for `material-ui` instead directly import the `TextWidget` from the source code hierarchy. As a result, because I've created a custom extension of the `TextWidget` I now need to create custom replacements for ALL of the `TextWidget` wrappers (i.e. `EmailWidget`, `PasswordWidget`, etc).

If the `material-ui` implementation were to pull the `TextWidget` from the registry instead of directly importing it, then I would not need to override every subclass.

### Version
`v3.0.2`

**Comment by epicfaace** [comment]
Yeah, that would be a good change. Do you know if this is the case for all other themes as well? Would you be willing to add a PR to fix it?

**Comment by heath-freenome** [comment]
I haven't looked at the other themes... I can try pushing a PR... at least for the source changes in material-ui

**Comment by heath-freenome** [comment]
It seems to be similar for `EmailWidget` in the two other typescript libraries (bootstrap-4 and fluent)...

**Comment by heath-freenome** [comment]
Wow... that's a lot of vulnerabilities:
```
react-jsonschema-form(master|✔) % npm install

> react-jsonschema-form@2.0.0-alpha.1 postinstall /Users/heath/dev/public/react-jsonschema-form
> lerna bootstrap

lerna notice cli v3.18.4
lerna info Bootstrapping 7 packages
lerna info Installing external dependencies
lerna info Symlinking packages and binaries
lerna success Bootstrapped 7 packages
added 62 packages from 37 contributors, removed 12 packages, updated 47 packages and audited 903 packages in 156.649s

11 packages are looking for funding
  run `npm fund` for details

found 408 vulnerabilities (357 low, 33 moderate, 18 high)
  run `npm audit fix` to fix them, or `npm audit` for details
```

**Comment by heath-freenome** [comment]
So the bootstrap4 changes were causing me too many problems and I aborted on them. I'll let somebody else figure it out.

**Comment by heath-freenome** [comment]
Also, what's the command to regenerate the snapshots for `material-ui`?

**Comment by epicfaace** [comment]
You can run `npm run test -- -u` to update snapshots.

--
Ashwin Ramaswami

**Comment by heath-freenome** [comment]
@epicfaace I have a PR but can't push it... What is the process?
```
remote: Permission to rjsf-team/react-jsonschema-form.git denied to heath-freenome.
fatal: unable to access 'https://github.com/rjsf-team/react-jsonschema-form.git/': The requested URL returned error: 403
```

**Comment by epicfaace** [comment]
You should fork this repo to create your own version (which will likely be
under https://github.com/heath-freenome/react-jsonschema-form.git/). Then,
you push to your fork and create a PR from your fork.

--
Ashwin Ramaswami
